app/models/claim/scrubber.rb

```diff
@@ -0,0 +1,47 @@
+# Removes attributes from a claim and its amendments
+class Claim
+  class Scrubber
+    def self.scrub!(claim, attributes_to_delete)
+      new(claim, attributes_to_delete).scrub!
+    end
+
+    attr_reader :claim, :attributes_to_delete
+
+    def initialize(claim, attributes_to_delete)
+      @claim = claim
+      @attributes_to_delete = attributes_to_delete.map(&:to_s)
+    end
+
+    def scrub!
+      ApplicationRecord.transaction do
+        claim.amendments.each { |amendment| scrub_amendment!(amendment) }
+        scrub_claim!
+        scrub_session!
+      end
+    end
+
+    private
+
+    def scrub_amendment!(amendment)
+      amendment_data_to_scrub = attributes_to_delete & amendment.claim_changes.keys.map(&:to_s)
+      personal_data_mask = amendment_data_to_scrub.to_h { |attr| [attr, nil] }
+      amendment.claim_changes.merge!(personal_data_mask)
+      amendment.personal_data_removed_at = Time.zone.now
+      amendment.save!
+    end
+
+    def scrub_claim!
+      personal_data_mask = attributes_to_delete.to_h { |attr| [attr, nil] }
+      attributes_to_set = personal_data_mask.merge(
+        personal_data_removed_at: Time.zone.now
+      )
+      claim.update_columns(attributes_to_set)
+    end
+
+    def scrub_session!
+      return unless claim.journey_session
+
+      claim.journey_session.update!(answers: {})
```

Comment:
are we sure we want to remove all answers and not just the answers that need scrubbing?

one example i can think of for FE is that claimants are supposed to have a teaching qualification or on course to attain a teaching qualification. this answer may want to be kept around so this can be verified at a future date

Reply:
should be fine as we use a different implementation there i guess, since for this change technically only applies to IRP

Reply:
> are we sure we want to remove all answers and not just the answers that need scrubbing?
I think it's safe as the answers should be copied to the claim upon submission. We probably will want to move scrubbing the session answers at the point of submission/claim creation instead, I've added scrubbing the session here in case we decide not to do that